odiglet/cmd/main.go

```diff
@@ -106,25 +81,102 @@ func main() {
 		os.Exit(-1)
 	}
 
-	err = kube.StartManager(ctx, mgr)
-	if err != nil {
-		log.Logger.Error(err, "Failed to start controller-runtime manager")
-		os.Exit(-1)
+	return &odiglet{
+		clientset:     clientset,
+		mgr:           mgr,
+		ctx:           ctx,
+		ebpfDirectors: ebpfDirectors,
 	}
+}
 
-	<-ctx.Done()
-	for _, director := range ebpfDirectors {
+func (o *odiglet) run() {
```

Comment:
I think it's a bit non-conventional to have the `ctx` as property.

Consider sending it to the `run` function

Reply:
Right, currently I keep it that way since we have the `initEbpf` part which gets a context, but that will be removed soon.